config/FindEOSPAC.cmake

```diff
@@ -5,81 +5,201 @@
+#   endif()
+#   # We might also try scraping the directory name for a regex match
+#   # "eospac-X.X.X"
+#   if( NOT EOSPAC_MAJOR )
+#     string( REGEX REPLACE ".*eospac-([0-9]+).([0-9]+).([0-9]+).*" "\\1"
+#       EOSPAC_MAJOR ${EOSPAC_INCLUDE_DIR} )
+#     string( REGEX REPLACE ".*eospac-([0-9]+).([0-9]+).([0-9]+).*" "\\2"
+#       EOSPAC_MINOR ${EOSPAC_INCLUDE_DIR} )
+#     string( REGEX REPLACE ".*eospac-([0-9]+).([0-9]+).([0-9]+).*" "\\3"
+#       EOSPAC_SUBMINOR ${EOSPAC_INCLUDE_DIR} )
+#   endif()
+# endif()
 
-# handle the QUIETLY and REQUIRED arguments and set EOSPAC_FOUND to TRUE if 
-# all listed variables are TRUE
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(EOSPAC DEFAULT_MSG EOSPAC_INCLUDE_DIR EOSPAC_LIBRARY)
-
-if (EOSPAC_FOUND)
-   set(EOSPAC_FOUND ${EOSPAC_FOUND} CACHE BOOL "Did we find the EOSPAC libraries?")
-   set(EOSPAC_INCLUDE_DIRS ${EOSPAC_INCLUDE_DIR})
-   set(EOSPAC_LIBRARIES    ${EOSPAC_LIBRARY} CACHE
-      FILEPATH "EOSPAC libraries for linking."  )
-   
-   string( REPLACE "_dll.lib" ".dll" EOSPAC_DLL ${EOSPAC_LIBRARY} )
-   mark_as_advanced( EOSPAC_DLL )
-   if( EXISTS ${EOSPAC_DLL} )
-      set(EOSPAC_DLL_LIBRARIES "${EOSPAC_DLL}" CACHE STRING 
-         "list of eospac dll files.")
-   mark_as_advanced( EOSPAC_DLL_LIBRARIES EOSPAC_LIBRARIES )
-   else()
-      set( EOSPAC_DLL "NOTFOUND")
-   endif()
+#=============================================================================
+# handle the QUIETLY and REQUIRED arguments and set EOSPAC_FOUND to TRUE if
+# all listed variables are TRUE.
+find_package_handle_standard_args( EOSPAC
+  FOUND_VAR
+    EOSPAC_FOUND
+  REQUIRED_VARS
+    EOSPAC_INCLUDE_DIR
+    EOSPAC_LIBRARY
+  VERSION_VAR
+    EOSPAC_VERSION
+    )
+
+mark_as_advanced( EOSPAC_ROOT_DIR EOSPAC_VERSION EOSPAC_LIBRARY
+  EOSPAC_INCLUDE_DIR EOSPAC_LIBRARY_DEBUG EOSPAC_USE_PKGCONFIG EOSPAC_CONFIG )
+
+#=============================================================================
+# Register imported libraries:
+# 1. If we can find a Windows .dll file (or if we can find both Debug and
+#    Release libraries), we will set appropriate target properties for these.
+# 2. However, for most systems, we will only register the import location and
+#    include directory.
+
+# Look for dlls, or Release and Debug libraries.
+if(WIN32)
+  string( REPLACE ".lib" ".dll" EOSPAC_LIBRARY_DLL
+    "${EOSPAC_LIBRARY}" )
+  string( REPLACE ".lib" ".dll" EOSPAC_LIBRARY_DEBUG_DLL
+    "${EOSPAC_LIBRARY_DEBUG}" )
 endif()
 
-if( VERBOSE )
-message("
-EOSPAC_FOUND =        ${EOSPAC_FOUND}
-EOSPAC_INCLUDE_DIRS = ${EOSPAC_INCLUDE_DIR}
-EOSPAC_LIBRARIES    = ${EOSPAC_LIBRARY}
-")
+if( EOSPAC_FOUND AND NOT TARGET EOSPAC::eospac )
+  if( EXISTS "${EOSPAC_LIBRARY_DLL}" )
+
+    # Windows systems with dll libraries.
+    add_library( EOSPAC::eospac SHARED IMPORTED )
+
+    # Windows with dlls, but only Release libraries.
```

Comment:
In the `find_library` command for `EOSPAC_LIBRARY` above, the path suffixes include `Debug`.  If a release version was not found, would the following `set_target_properties` exclude the Debug library?

Reply:
For Makefile builds, there will only be one library.  The Debug/Release subfolders are typically used by multi-config build tools like Visual Studio or Xcode.  These build tools know what to do with the Debug/Release information.